trie/database.go

```diff
@@ -17,11 +17,18 @@
 package trie
 
 import (
+	"bytes"
+	"fmt"
+	"io"
 	"sync"
 
+	scale "github.com/ChainSafe/gossamer/codec"
+	"github.com/ChainSafe/gossamer/common"
 	"github.com/ChainSafe/gossamer/polkadb"
 )
 
+var LATEST_HASH_KEY = []byte("latest_hash")
+
 // StateDB is a wrapper around a polkadb
 type StateDB struct {
```

Comment:
This should get updated and used. It should probably be called `Database`. I think this should provide `LoadRootHash()`,  `SaveRootHash()` and anything else that is not direclty relevant for the Trie to function. We can probably also remove most of its fields.

The reason we want to make sure to have an extra layer between the trie and DB is so we can implement caching further down the road.

Reply:
okay, the issue with LoadRootHash from the DB is we would have to pass in the root hash, we can't get it directly from the trie. also the other services are interacting with the trie only, I think it's a bit nicer to just have it accessible from the trie rather than doing trie.Db.stuff

Reply:
I have made the requested changes upon discussion, let me know what you think